core/include/traccc/finding/combinatorial_kalman_filter_algorithm.hpp

```diff
@@ -28,12 +28,14 @@ namespace traccc::host {
 class combinatorial_kalman_filter_algorithm
     : public algorithm<track_candidate_container_types::host(
           const default_detector::host&,
-          const detray::bfield::const_field_t::view_t&,
+          const detray::bfield::const_field_t<
+              telescope_detector::host::scalar_type>::view_t&,
```

Comment:
```suggestion
              default_detector::host::scalar_type>::view_t&,
```

Since it's "just" the scalar type that's being given as a template, I wonder if there may be a better way of providing it to the b-field type. 🤔

Reply:
I am not sure. What is the plan for the algebra plugins integration in traccc long term? Will we stay with the centrally defined algebra types? Then we could just do the same with the B-field types.  We will eventually need more types than just the constant field here, right?

Reply:
Since Detray already started to define some b-field typedefs, we could just leave them there. But yes, we will definitely need at least a "parametrized" one (which loads a map from a file and interpolates between the values) and I guess some "functional" one, which would allow using some user function for calculating the magnetic field at position X-Y-Z.

@stephenswat will have to advise about the exact type names on those. (And whether the latter is even possible/practical.)